include/Common/Exceptions.h

```diff
@@ -29,7 +29,10 @@ struct ObjectNotFoundError : virtual ExceptionBase
 {
   const char *what() const noexcept override
   {
-    return "Object not found error";
+    std::string message = "Object not found: ";
+    auto* errinfo = boost::get_error_info<errinfo_object_name>(*this);
+    message += errinfo ? *boost::get_error_info<errinfo_object_name>(*this) : "(object_name not specified)";
```

Comment:
is it possible to simply use `*errinfo` instead of calling `get_error_info` again?

Reply:
of course, I must have been tired. thanks